variables.tf

```diff
@@ -83,10 +97,10 @@ variable "tagName" {
   description = "Name tag for the servers"
 }
 
-variable "vpc_id" {
-  description = "The AWS VPC ID which you want to deploy your instances"
+variable "user_data" {
+  default = ""
 }
 
-variable "associate_public_ip_address" {
-  default = false
+variable "vpc_id" {
```

Comment:
Hmmm, mixed feelings on this, especially since I'm assuming that many of the inputs to this module will be outputs from the [VPC module, or a similar modules outputs.]( https://github.com/terraform-community-modules/tf_aws_vpc#outputs ) which doesn't output a `name` variable.  I could be convinced, but thats probably a parking lot conversation :)